src/mappings-decoder.ts

```diff
@@ -24,22 +25,22 @@ export default class MappingsDecoder {
     this.delegate = delegate;
   }
 
-  decode(reader) {
-    while (reader.ptr < reader.limit) {
-      switch (reader.buf[reader.ptr]) {
+  decode(reader: Reader) {
+    while (reader.hasNext()) {
+      switch (reader.peek()) {
```

Comment:
Loops can be weird, especially with switches, please trace in v8 to see how this loop is handled